src/messageschannel.cpp

```diff
@@ -152,26 +152,33 @@ void MatrixMessagesChannel::processMessageEvent(const QMatrixClient::RoomMessage
     qDebug().noquote() << Q_FUNC_INFO << "Process message" << doc.toJson(QJsonDocument::Indented);
     Tp::MessagePart header;
     header[QStringLiteral("message-token")] = QDBusVariant(event->id());
-    header[QStringLiteral("message-sent")]  = QDBusVariant(event->timestamp().toMSecsSinceEpoch() / 1000);
-    header[QStringLiteral("message-received")] = QDBusVariant(event->timestamp().toMSecsSinceEpoch() / 1000);
     if (event->senderId() == m_connection->matrix()->user()->id()) {
-        header[QStringLiteral("message-sender")]   = QDBusVariant(m_connection->selfHandle());
+        header[QStringLiteral("message-sender")]    = QDBusVariant(m_connection->selfHandle());
         header[QStringLiteral("message-sender-id")] = QDBusVariant(m_connection->selfID());
     } else {
-        header[QStringLiteral("message-sender")]   = QDBusVariant(m_connection->ensureContactHandle(event->senderId()));
+        header[QStringLiteral("message-sender")]    = QDBusVariant(m_connection->ensureContactHandle(event->senderId()));
         header[QStringLiteral("message-sender-id")] = QDBusVariant(event->senderId());
     }
+    header[QStringLiteral("message-sent")]  = QDBusVariant(event->timestamp().toMSecsSinceEpoch() / 1000);
+    header[QStringLiteral("message-received")] = QDBusVariant(event->timestamp().toMSecsSinceEpoch() / 1000);
+    if (event->isRedacted()) {
+        header[QStringLiteral("message-type")]    = QDBusVariant(Tp::ChannelTextMessageTypeDeliveryReport);
```

Comment:
A delivery report is just a 'report' and does not present a message. Probably we want the type to be 'normal' to make the client add the message as an incoming or outgoing message though it is already known to be deleted.

The next part would be to implement deleted messages on the client (UI) side.

Reply:
thank you! propably I have same issue with `MatrixMessagesChannel::onPendingEventChanged`